application-analytics-default/src/main/java/com/xwiki/analytics/script/AnalyticsScriptService.java

```diff
@@ -71,6 +85,40 @@ public JsonNode makeRequest(Map<String, String> parameters, Map<String, String>
         }
     }
 
+    /**
+     * Endpoint for all the aggregator jobs.
+     *
+     * @param hint for which aggregator you want to use
+     */
+    public void aggregate(String hint)
+    {
+
+        for (Aggregator aggregator : aggregators.get()) {
+            if (hint.equals(aggregator.getHint())) {
+                aggregator.aggregateData();
```

Comment:
Why do it like this instead of something like `componentManager.getInstance(hint)`?

Reply:
Maybe you should also check the rights of the user that executes this script?

Reply:
The decision to avoid using componentManager.getInstance(hint) was based on discussions I had with @oanalavinia while implementing the Matomo manager:

    [Discussion 1](https://github.com/xwikisas/application-analytics/pull/12#discussion_r1289797242)
    [Discussion 2](https://github.com/xwikisas/application-analytics/pull/23/files#r1318583561)